lib/sumo/client.rb

```diff
@@ -32,13 +49,27 @@ def request(hash, &block)
   # Private functions that operate on the request and response.
 
   def add_defaults(hash)
-    hash.merge(
-      :headers => default_headers.merge(hash[:headers] || {}),
-      :path => "/api/v#{Sumo::API_VERSION}#{hash[:path]}"
-    )
+    hash[:headers] = default_headers.merge(hash[:headers] || {})
+    hash[:path] = "/api/v#{Sumo::API_VERSION}#{hash[:path]}" unless
+      hash[:path].index("/api/v#{Sumo::API_VERSION}") == 0
+    hash
   end
   private :add_defaults
 
+  # Recursively handle redirection up to 10 level depth
+  def handle_redirect(response, hash, &block)
+    fail 'Too many redirects' if @redirect_depth > 9
+
+    endpoint = response.headers['Location']
+               .match(%r{^(https://.+\.[a-z]+)/}).to_a[1]
+
+    @redirect_depth += 1
```

Comment:
I would feel better if `@redirect_depth` wasn't an instance variable, since it really should happen on a request-per-request basis. Perhaps `#handle_request` could accept another parameter, `redirect_depth`, which defaults to `0` and is increased on every call that `#handle_redirect` makes to the method.